geode-gfsh/src/main/java/org/apache/geode/management/internal/cli/GfshParser.java

```diff
@@ -122,21 +122,30 @@ else if (insideQuoteOf == c) {
     List<String> splitWithWhiteSpaces = splitWithWhiteSpace(userInput);
 
     List<String> furtherSplitWithEquals = new ArrayList<>();
+    boolean doNotRemoveEqualsInNextToken = false;
     for (String token : splitWithWhiteSpaces) {
-      // if this token has equal sign, split around the first occurrence of it
-      int indexOfFirstEqual = token.indexOf('=');
-      if (indexOfFirstEqual < 0) {
+      if (doNotRemoveEqualsInNextToken) {
         furtherSplitWithEquals.add(token);
+        doNotRemoveEqualsInNextToken = false;
       } else {
-        String left = token.substring(0, indexOfFirstEqual);
-        String right = token.substring(indexOfFirstEqual + 1);
-        if (left.length() > 0) {
-          furtherSplitWithEquals.add(left);
-        }
-        if (right.length() > 0) {
-          furtherSplitWithEquals.add(right);
+        // if this token has equal sign, split around the first occurrence of it
+        int indexOfFirstEqual = token.indexOf('=');
+        if (indexOfFirstEqual < 0) {
+          furtherSplitWithEquals.add(token);
+        } else {
+          String left = token.substring(0, indexOfFirstEqual);
+          String right = token.substring(indexOfFirstEqual + 1);
+          if (left.length() > 0) {
+            furtherSplitWithEquals.add(left);
+          }
+          if (right.length() > 0) {
+            furtherSplitWithEquals.add(right);
+          }
         }
       }
+      if (token.equals("--query") || token.equals("--J")) {
```

Comment:
anyway to not hardcode these two conditions? maybe based the decision on "do not remove = sign if it's in the quotes"?

Reply:
good point, I have changed the solution. Thanks!